src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,195 @@
+                    bsoncxx::document::value doc_update1 = bsoncxx::builder::stream::document{}
+                        << "$inc" << bsoncxx::builder::stream::open_document
+                            << "n" << -amount
+                        << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    config->collection.update_one(doc_filter1.view(), doc_update1.view(),
+                                                  config->optionsUpdate);
+
+                    // result2 = db.hltest.find_one( { '_id': 2 }, session=session )
+                    bsoncxx::document::value doc_filter2 = bsoncxx::builder::stream::document{}
+                        << "_id" << 2
+                        << bsoncxx::builder::stream::finalize;
+                    auto result2 = config->collection.find_one(doc_filter2.view(), config->optionsFind);
+
+                    // db.hltest.update_one( {'_id': 2}, {'$inc': {'n': amount}}, session=session )
+                    bsoncxx::document::value doc_update2 = bsoncxx::builder::stream::document{}
+                        << "$inc" << bsoncxx::builder::stream::open_document
+                            << "n" << amount
+                        << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    config->collection.update_one(doc_filter2.view(), doc_update2.view(),
+                                                  config->optionsUpdate);
+
+                    // result = db.hltest.aggregate( [ { '$group': { '_id': 'foo', 'total' : { '$sum': '$n' } } } ], session=session ).next()
+                    mongocxx::pipeline p{};
+                    bsoncxx::document::value doc_group = bsoncxx::builder::stream::document{}
+                            << "_id" << "foo"
+                            << "total" << bsoncxx::builder::stream::open_document
+                                << "$sum" << "$n"
+                            << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    p.group(doc_group.view());
+                    auto cursor = config->collection.aggregate(p, config->optionsAggregate);
+                    // Check for isolation or consistency errors
+                    // Note: If there's skew in the sum total, we never fix it. So after the first
+                    // error, this counter is likely to increment on every loop after that. The
+                    // exact value of the metric is therefore uninteresting, it's only significant
+                    // whether it is zero or non-zero.
+                    for (auto doc : cursor) {
+                        if (doc["total"].get_int64() != 200) {
+                            ctx.addErrors(1);
+                        }
+                    }
```

Comment:
you may also want to `ctx.addDocs(1)` for each doc and `ctx.addBytes(doc.size())`. These can serve as a sort of canary that the behavior of the test (number of bytes/docs) hasn't changed dramatically from run-over-run.

Reply:
Intended use case is to run exactly 500 repeats, so it felt more noisy than useful to keep those.